org.mixedrealitytoolkit.core/Utilities/Extensions/ComponentExtensions.cs

```diff
@@ -48,8 +48,7 @@ public static T FindAncestorComponent<T>(this Component component, bool includeS
         /// </remarks>
         public static T EnsureComponent<T>(this GameObject gameObject) where T : Component
         {
-            T foundComponent = gameObject.GetComponent<T>();
-            return foundComponent == null ? gameObject.AddComponent<T>() : foundComponent;
+            return gameObject.TryGetComponent(out T foundComponent) ? foundComponent : gameObject.AddComponent<T>();
```

Comment:
Thank you for this! I made the same change in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/1004/files#diff-9827b8e3e0230bbf3ccacee31f3b8a03d51c41e428ff954d60a08699353580f1R51, which is still pending review, so I'm happy to go ahead and merge yours!